### ISSUE TITLE
saveWidget and background-color

### ISSUE THREAD

**Issue by ekstroem** [ramnathv/htmlwidgets#119]
Hi all,

As a sort of follow-up to #113 I'm trying to include a DiagrammeR graph into a slidify (impressjs framework) and am using `saveWIdget` to save the widget before including it into my markdown file.

The html-code produced by saveWidget has
```
<body style="background-color:white;">
```

Would it be possible to change that to 
```
<body style="background-color:transparent;">
```
either permanently in `toHTML` or at least allow an argument to change it directly in the file. Unless it breaks something I'm not aware of.

[ This only fixes my problem partly with the impressjs framework for slidify. Things are perfect with revealjs but somehow the background keeps being white with impressjs. However, by setting the style the bounding box around the diagram at least becomes transparent ]

**Comment by ramnathv** [comment]
Can you move to this to the `slidify` repo as it is a `slidify` specific issue?

**Comment by ekstroem** [comment]
Just a brief follow-up on this. It *is* an `htmlwidget` issue and not a slidify issue since my feature request/problem lies in the call to `htmltools:save_html` in the `saveWidget` function. A simple fix would be to update saveWidget to 

```
function (widget, file, selfcontained = TRUE, libdir = NULL, background = "transparent") 
{
    html <- toHTML(widget, standalone = TRUE)
    if (is.null(libdir)) {
        libdir <- paste(tools::file_path_sans_ext(basename(file)), 
            "_files", sep = "")
    }
    htmltools::save_html(html, file = file, libdir = libdir, background=background)
    if (selfcontained) {
        if (!pandoc_available()) {
            stop("Saving a widget with selfcontained = TRUE requires pandoc. For details see:\n", 
                "https://github.com/rstudio/rmarkdown/blob/master/PANDOC.md")
        }
        pandoc_self_contained_html(file, file)
        unlink(libdir, recursive = TRUE)
    }
    invisible(NULL)
}
```

(or possibly `background="white"` as the default argument to `saveWidget`) but I prefer the transparent version.

Want me to make a pull request?

[ Oh and I fixed the slidify context issue as long as the background can be set like this, but that is just icing on the cake ]

**Comment by jjallaire** [comment]
Yes, a pull request would be great, but let's keep the default white (as
for some widgets transparent produces an unworkable result).

**Comment by ramnathv** [comment]
@ekstroem Thanks. I realized it is a `htmlwidgets` issue and deleted my earlier comment. Thanks for the PR.

**Comment by timelyportfolio** [comment]
This should probably also be included in #58 and #68

**Comment by ekstroem** [comment]
@timelyportfolio Yup. I'm using a hacked version of `as.iframe` myself that includes the background but I wasn't sure what version (if any) of `as.iframe´ that would end up being included in the project.
